common/net/configure_port.go

```diff
@@ -74,6 +74,19 @@ func (lc ListenRangeConfig) Listen(ctx context.Context) (*Listener, error) {
 
 	err := retry.Config{
 		RetryDelay: func() time.Duration { return 1 * time.Millisecond },
+		ShouldRetry: func(err error) bool {
```

Comment:
One simplification would be to just`return portRange > 0`.

Also if there is no port range may be we could use a linearly growing retry delay instead.

Reply:
Would there be a time when the port would come available if we waited long enough?

Reply:
oh... I guess when the previous build completes its boot command.

Reply:
Yes